src/services/events.getTrainingReportAlerts.test.js

```diff
@@ -272,14 +262,251 @@ describe('getTrainingReportAlerts', () => {
 
     it('fetches the correct alerts', async () => {
       const alerts = await getTrainingReportAlerts();
-      const idsToCheck = alerts.map((i) => i.id).filter((i) => testData.ids.has(i));
-
-      expect(idsToCheck.sort()).toStrictEqual([
-        ...testData.ist.missingEventInfo,
-        ...testData.ist.missingSessionInfo,
-        ...testData.ist.noSessionsCreated,
-        ...testData.ist.eventNotCompleted,
-      ].sort());
+      // Filter to just time-based alerts from our test data
+      const timeBasedAlerts = alerts.filter(
+        (a) => !['waitingForApproval', 'changesNeeded'].includes(a.alertType),
+      );
+      const idsToCheck = timeBasedAlerts
+        .map((i) => i.id)
+        .filter((i) => testData.ids.has(i));
+      const expectedIds = [
+        ...new Set([
+          ...testData.ist.missingEventInfo,
+          ...testData.ist.missingSessionInfo,
+          ...testData.ist.noSessionsCreated,
+          ...testData.ist.eventNotCompleted,
+        ]),
+      ];
+
+      // Verify all expected IDs are present in time-based alerts
+      expectedIds.forEach((expectedId) => {
+        expect(idsToCheck).toContain(
+          expectedId,
+          `Expected ${expectedId} to be in alerts for time-based alert types`,
+        );
+      });
+    });
+
+    it('triggers missingEventInfo alert 20 days after END date', async () => {
+      // Create event with endDate 20 days ago
+      const eventWithOldEndDate = await EventReportPilot.create({
+        ownerId,
+        collaboratorIds: [faker.datatype.number()],
+        pocIds: [faker.datatype.number()],
+        regionId: 1,
+        data: {
+          eventName: 'Missing Event Info Test',
+          eventId: `R01-TR-ENDDATE-${ownerId}`,
+          status: TRAINING_REPORT_STATUSES.IN_PROGRESS,
+          trainingType: 'Series',
+          targetPopulations: ['Children & Families'],
+          reasons: ['Coaching'],
+          vision: 'Testing end date!',
+          eventSubmitted: false,
+          startDate: new Date(new Date().setMonth(new Date().getMonth() - 2)),
```

Comment:
Having all these `new Date()`'s one after the other looks slightly unnecessary to me. Could you just make a `currentDate` const and then do all of the setDates and getDates based off of that? Then again I suppose this is just for a test file so no biggie either way.